File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/ServicePlanResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import com.ssafy.e106.domain.subscription.entity.ServicePlan;

public record ServicePlanResponse(
    Long servicePlanId,
    String planName,
    String billingCycle,
    Integer monthlyPrice) {

  public static ServicePlanResponse from(ServicePlan servicePlan) {
    return new ServicePlanResponse(
        servicePlan.getServicePlanId(),
        servicePlan.getPlanName(),
        servicePlan.getBillingCycle(),
        servicePlan.getMonthlyPrice());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/service/ServicePlanResponse.kt
package com.ssafy.e106.data.dto.service

import kotlinx.serialization.Serializable

@Serializable
data class ServicePlanResponse(
    val servicePlanId: Long,
    val planName: String,
    val billingCycle: String,
    val monthlyPrice: Int,
)
